icu4c/source/i18n/complexunitsconverter.cpp

```diff
@@ -0,0 +1,86 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+#include <utility>
+
+#include "cmemory.h"
+#include "complexunitsconverter.h"
+#include "uassert.h"
+#include "unicode/fmtable.h"
+#include "unitconverter.h"
+
+U_NAMESPACE_BEGIN
+
+ComplexUnitsConverter::ComplexUnitsConverter(const MeasureUnit inputUnit, const MeasureUnit outputUnits,
+                                             UErrorCode &status) {
+    auto singleUnits = outputUnits.splitToSingleUnits(status);
+    MaybeStackVector<MeasureUnit> singleUnitsInOrder;
+    for (int i = 0, n = singleUnits.length(); i < n; ++i) {
+        // TODO(younies): ensure units being in order in phase 2. Now, the units in order by default.
+        singleUnitsInOrder.emplaceBack(singleUnits[i]);
```

Comment:
Here and elsewhere: you need to check that the return value of `emplaceBack` is not nullptr.  If it is nullptr, you need to set a U_MEMORY_ALLOCATION_ERROR.

Reply:
Done by adding `MayBeStackVector::emplaceBackAndConfirm`